experimental/printer/dom/chunk.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2020-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dom
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/rivo/uniseg"
+)
+
+const (
+	space = " "
+)
+
+// TODO: considering whether the panics in Chunk are too strict... I am of the opinion that
+// we should be inflexible with this behaviour because we want this API to inform the use
+// patterns strictly, but open to the opinions of others.
+//
+// TODO: do we need some kind of Zero concept for Chunk (rather than just relying on nil checks)?
```

Comment:
I'm not entirely confident on the behaviour and API of `Chunk`. I have a couple of panics to enforce some patterns, but I'm not sure if they are set at the correct layer. I also find that there are several `nil` checks for `Chunk` and `Chunks`, and I'm not sure if this should have a `Zero` type on `Chunk`.